src/3p-analytics-common.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** @enum {string} */
+export const AMP_ANALYTICS_3P_MESSAGE_TYPE = {
+  READY: 'R',
+  CREATIVE: 'C',
+  EVENT: 'E',
+  RESPONSE: 'A',
```

Comment:
The basic messaging flow is:
1. xframe sends message informing the parent page that it is ready (R). This /could/ be done with the page load callback, but Keith or Justin, I forget which, prefers sending a message instead. This is because we can send a message can happen sooner, i.e. doesn't necessarily require every other (unrelated) resource to be loaded first
2. Creative sends extra config data to xframe (C). Since multiple creatives may use the same xframe, it doesn't make sense to specify this on the URL/name attr/etc. (We could save a few ms for the first creative to use the xframe I suppose, but another method is definitely required for subsequent creatives to say "here is my extra config data")
3. Creative sends events (E) to xframe
4. After some amount of time, or after some number of events, xframe may choose to send a response (A, since R was already taken) to the creative.